express-notes-api/index.js

```diff
@@ -0,0 +1,166 @@
+const express = require("express")
+const app = express()
+
+const fs = require("fs")
+
+const data = require("./data.json")
+let {nextId,notes}=data
+
+app.use(express.json())
+
+app.get("/api/notes",(req,res)=>{
+  let notesArr = []
+
+  for (const prop in notes){
+    notesArr.push(notes[prop])
+  }
+
+  res.status(200).json(notesArr)
+})
+
+app.get("/api/notes/:id",(req,res)=>{
+  if (isNaN(req.params.id) === false && req.params.id > 0 && Number.isInteger(parseFloat(req.params.id))){
+    let id = parseInt(req.params.id)
+
+    const idArr=[]
+    for (const prop in notes) {
+      idArr.push(parseInt(prop))
+    }
+
+    if(idArr.includes(id)){
+      res.status(200).json(notes[id])
+
+    } else {
+      const err2 = {
+        "error": `cannot find note with id ${id}`
+      }
+      res.status(404).json(err2)
+    }
+
+  } else {
+    const err = {
+      "error": "id must be a positive integer"
+    }
+    res.status(400).json(err)
+  }
+})
+
+
+app.post("/api/notes",(req,res)=>{
+  let newNote=req.body
```

Comment:
here you should be using req.body.content. you can't guarantee that the body will just be a single property